<!DOCTYPE html><html><head><title>R: Expected Value of Scaled Unit Deviance for Linear Exponential...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script>
const macros = { "\\R": "\\textsf{R}", "\\mbox": "\\text", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css">
</head><body><div class="container"><main>

<table style="width: 100%;"><tr><td>expectedDeviance {statmod}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Expected Value of Scaled Unit Deviance for Linear Exponential Families</h2>

<h3>Description</h3>

<p>Expected value and variance of the scaled unit deviance for common generalized linear model families.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expectedDeviance(mu, family="binomial", binom.size, nbinom.size, gamma.shape)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mu">mu</code></td>
<td>
<p>numeric vector or matrix giving mean of response variable.</p>
</td></tr>
<tr><td><code id="family">family</code></td>
<td>
<p>character string indicating the linear exponential family. Possible values are <code>"binomial"</code>,<code>"gaussian"</code>, <code>"Gamma"</code>, <code>"inverse.gaussian"</code>, <code>"poisson"</code> or <code>"negative.binomial"</code>.</p>
</td></tr>
<tr><td><code id="binom.size">binom.size</code></td>
<td>
<p>integer vector giving the number of binomial trials when <code>family = "binomial"</code>. Equivalent to the <code>"size"</code> argument of <code>pbinom</code>.</p>
</td></tr>
<tr><td><code id="nbinom.size">nbinom.size</code></td>
<td>
<p>numeric vector giving the negative binomial size parameter when <code>family = "negative.binomial"</code>, such that the variance of the response variable is <code>mu + mu^2 / nbinom.size</code>. Equivalent to the <code>"size"</code> parameter of <code>pnbinom</code>.</p>
</td></tr>
<tr><td><code id="gamma.shape">gamma.shape</code></td>
<td>
<p>numeric vector giving the gamma shape parameter when <code>family = "Gamma"</code>, such that the variance of the response variable is <code>mu^2 / gamma.shape</code>. Equivalent to the <code>"shape"</code> parameter of <code>pgamma</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a generalized linear model (GLM), the scaled unit deviances can be computed using <code>d &lt;- f$dev.resids(y, mu, wt=1/phi)</code> where <code>f</code> is the GLM family object, <code>y</code> is the response variable, <code>mu</code> is the vector of means and <code>phi</code> is the vector of GLM dispersions (incorporating any prior weights).
</p>
<p>The scaled unit deviances are often treated as being chiquare distributed on 1 df, so the mean should be 1 and the variance should be 2.
This distribution result only holds however when the saddlepoint approximation is accurate for the response variable distribution (Dunn and Smyth, 2018).
In other cases, the expected value and variance of the unit deviances can be far from the nominal values.
The <code>expectedDeviance</code> function returns the exact mean and variance of the unit deviance for the usual GLM familes assuming that <code>mu</code> is the true mean and <code>phi</code> is the true dispersion.
</p>
<p>When <code>family</code> is <code>"poisson"</code>, <code>"binomial"</code> or <code>"negative.binomial"</code>, the expected values and variances are computed using Chebyshev polynomial approximations.
When <code>family = "Gamma"</code>, the function uses exact formulas derived by Smyth (1989).
</p>
<p>Exact calculation of expected values and variances of negative binomial unit deviances have been used to construct adjusted residual deviances and adjusted residual degrees of freedom when fitting negative binomial generalized linear models in the edgeR software package (Baldoni et al 2024; Baldoni et al 2025; Chen et al, 2025).
</p>


<h3>Value</h3>

<p>A list with the components
</p>
<table role = "presentation">
<tr><td><code>mean</code></td>
<td>
<p>expected values</p>
</td></tr>
<tr><td><code>variance</code></td>
<td>
<p>variances</p>
</td></tr>
</table>
<p>both of which have the same length and dimensions as the input <code>mu</code>.
</p>


<h3>Author(s)</h3>

<p>Lizhong Chen and Gordon Smyth</p>


<h3>References</h3>

<p>Baldoni PL, Chen L, Smyth GK (2024).
Faster and more accurate assessment of differential transcript expression with Gibbs sampling and edgeR v4.
<em>NAR Genomics and Bioinformatics</em> 6(4), lqae151.
<a href="https://doi.org/10.1093/nargab/lqae151">doi:10.1093/nargab/lqae151</a>.
</p>
<p>Baldoni PL, Chen L, Li M, Chen Y, Smyth GK (2025).
Dividing out quantification uncertainty enables assessment of differential transcript usage with limma and edgeR.
<em>bioRxiv</em>
<a href="https://doi.org/10.1101/2025.04.07.647659">doi:10.1101/2025.04.07.647659</a>.
</p>
<p>Chen Y, Chen L, Lun ATL, Baldoni PL, Smyth GK (2025).
edgeR v4: powerful differential analysis of sequencing data with expanded functionality and improved support for small counts and larger datasets.
<em>Nucleic Acids Research</em> 53(2), gkaf018.
<a href="https://doi.org/10.1093/nar/gkaf018">doi:10.1093/nar/gkaf018</a>
</p>
<p>Dunn PK, Smyth GK (2018).
<em>Generalized linear models with examples in R</em>.
Springer, New York, NY.
<a href="https://doi.org/10.1007/978-1-4419-0118-7">doi:10.1007/978-1-4419-0118-7</a>
</p>
<p>Smyth, G. K. (1989).
Generalized linear models with varying dispersion.
<em>J. R. Statist. Soc. B</em>, <b>51</b>, 47-61.
<a href="https://doi.org/10.1111/j.2517-6161.1989.tb01747.x">doi:10.1111/j.2517-6161.1989.tb01747.x</a>
</p>


<h3>See Also</h3>

<p><code><a href="../../stats/html/family.html">family</a></code>, <code><a href="../../statmod/help/meanval.digamma.html">meanval.digamma</a></code>, <code><a href="../../statmod/help/d2cumulant.digamma.html">d2cumulant.digamma</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Poisson example
lambda &lt;- 3
nsim &lt;- 1e4
y &lt;- rpois(nsim, lambda=lambda)
d &lt;- poisson()$dev.resids(y=y, mu=rep(lambda,nsim), wt=1)
c(mean=mean(d), variance=var(d))
unlist(expectedDeviance(mu=lambda, family="poisson"))

# binomial example
n &lt;- 10
p &lt;- 0.01
y &lt;- rbinom(nsim, prob=p, size=n)
d &lt;- binomial()$dev.resids(y=y/n, mu=rep(p,nsim), wt=n)
c(mean=mean(d), variance=var(d))
unlist(expectedDeviance(mu=p, family="binomial", binom.size=n))

# gamma example
alpha &lt;- 5
beta &lt;- 2
y &lt;- beta * rgamma(1e4, shape=alpha)
d &lt;- Gamma()$dev.resids(y=y, mu=rep(alpha*beta,n), wt=alpha)
c(mean=mean(d), variance=var(d))
unlist(expectedDeviance(mu=alpha*beta, family="Gamma", gamma.shape=alpha))

# negative binomial example
library(MASS)
mu &lt;- 10
phi &lt;- 0.2
y &lt;- rnbinom(nsim, mu=mu, size=1/phi)
f &lt;- MASS::negative.binomial(theta=1/phi)
d &lt;- f$dev.resids(y=y, mu=rep(mu,nsim), wt=1)
c(mean=mean(d), variance=var(d))
unlist(expectedDeviance(mu=mu, family="negative.binomial", nbinom.size=1/phi))

# binomial expected deviance tends to zero for p small:
p &lt;- seq(from=0.001,to=0.11,len=200)
ed &lt;- expectedDeviance(mu=p,family="binomial",binom.size=10)
plot(p,ed$mean,type="l")
</code></pre>

<hr><div style="text-align: center;">[Package <em>statmod</em> version 1.5.1 <a href="00Index.html">Index</a>]</div></main>

</div>
</body></html>
