<!DOCTYPE html><html><head><title>R: Make a RangedSummarizedExperiment object from an...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script>
const macros = { "\\R": "\\textsf{R}", "\\mbox": "\\text", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css">
</head><body><div class="container"><main>

<table style="width: 100%;"><tr><td>makeSummarizedExperimentFromExpressionSet {SummarizedExperiment}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Make a RangedSummarizedExperiment object from an ExpressionSet and
vice-versa</h2>

<h3>Description</h3>

<p>Coercion between <a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> and
<a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> is supported in both directions.
</p>
<p>For going from <a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> to
<a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a>, the
<code>makeSummarizedExperimentFromExpressionSet</code> function is also
provided to let the user control how to map features to ranges.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeSummarizedExperimentFromExpressionSet(from,
                                          mapFun=naiveRangeMapper,
                                          ...)

## range mapping functions
naiveRangeMapper(from)
probeRangeMapper(from)
geneRangeMapper(txDbPackage, key = "ENTREZID")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="from">from</code></td>
<td>

<p>An <a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> object.
</p>
</td></tr>
<tr><td><code id="mapFun">mapFun</code></td>
<td>

<p>A function which takes an <a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> object and
returns a GRanges, or GRangesList object which
corresponds to the genomic ranges used in the ExpressionSet.  The
<a href="../../base/help/rownames.html">rownames</a> of the returned <a href="../../GenomicRanges/help/GRanges.html">GRanges</a>
are used to match the <a href="../../Biobase/help/featureNames.html">featureNames</a> of the
<a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a>.
</p>
<p>The <code>naiveRangeMapper</code> function is used by default.
</p>
</td></tr>
<tr><td><code id="...">...</code></td>
<td>

<p>Additional arguments passed to <code>mapFun</code>.
</p>
</td></tr>
<tr><td><code id="txDbPackage">txDbPackage</code></td>
<td>

<p>A character string with the Transcript Database to use for the mapping.
</p>
</td></tr>
<tr><td><code id="key">key</code></td>
<td>

<p>A character string with the Gene key to use for the mapping.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>makeSummarizedExperimentFromExpressionSet</code> takes an
<a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> object as input and a <em>range mapping
function</em> that maps the features to ranges.  It then returns a
<a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> object that corresponds to the input.
</p>
<p>The range mapping functions return a GRanges object, with the
<code>rownames</code> corresponding to the <a href="../../Biobase/help/featureNames.html">featureNames</a> of
the <a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> object.
</p>


<h3>Author(s)</h3>

<p>Jim Hester, <a href="james.f.hester@gmail.com">james.f.hester@gmail.com</a></p>


<h3>See Also</h3>


<ul>
<li> <p><a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> objects.
</p>
</li>
<li> <p><a href="../../Biobase/help/ExpressionSet.html">ExpressionSet</a> objects in the <span class="pkg">Biobase</span> package.
</p>
</li>
<li> <p><a href="../../GenomicFeatures/help/TxDb.html">TxDb</a> objects in the <span class="pkg">GenomicFeatures</span>
package.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## ---------------------------------------------------------------------
## GOING FROM ExpressionSet TO SummarizedExperiment
## ---------------------------------------------------------------------

data(sample.ExpressionSet, package="Biobase")

# naive coercion
makeSummarizedExperimentFromExpressionSet(sample.ExpressionSet)
as(sample.ExpressionSet, "RangedSummarizedExperiment")
as(sample.ExpressionSet, "SummarizedExperiment")

# using probe range mapper
makeSummarizedExperimentFromExpressionSet(sample.ExpressionSet, probeRangeMapper)

# using the gene range mapper
se &lt;- makeSummarizedExperimentFromExpressionSet(
    sample.ExpressionSet,
    geneRangeMapper("TxDb.Hsapiens.UCSC.hg19.knownGene")
)
se
rowData(se)  # duplicate row names

## ---------------------------------------------------------------------
## GOING FROM SummarizedExperiment TO ExpressionSet
## ---------------------------------------------------------------------

example(RangedSummarizedExperiment)  # to create 'rse'
rse
as(rse, "ExpressionSet")
</code></pre>

<hr><div style="text-align: center;">[Package <em>SummarizedExperiment</em> version 1.40.0 <a href="00Index.html">Index</a>]</div></main>

</div>
</body></html>
