<!DOCTYPE html><html><head><title>R: SummarizedExperiment objects</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script>
const macros = { "\\R": "\\textsf{R}", "\\mbox": "\\text", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css">
</head><body><div class="container"><main>

<table style="width: 100%;"><tr><td>SummarizedExperiment-class {SummarizedExperiment}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>SummarizedExperiment objects</h2>

<h3>Description</h3>

<p>The SummarizedExperiment class is a matrix-like container where rows
represent features of interest (e.g. genes, transcripts, exons, etc...)
and columns represent samples (with sample data summarized as a
DataFrame). A SummarizedExperiment object contains one or more
assays, each represented by a matrix-like object of numeric or other mode.
</p>
<p>Note that SummarizedExperiment is the parent of the
<a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> class which means that all the methods
documented below also work on a <a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## Constructor for RangedSummarizedExperiment/SummarizedExperiment objects

SummarizedExperiment(assays=SimpleList(),
                     rowData=NULL, rowRanges=NULL,
                     colData=DataFrame(),
                     metadata=list(),
                     checkDimnames=TRUE)

## Accessors

assayNames(x, ...)
assayNames(x, ...) &lt;- value
assays(x, withDimnames=TRUE, ...)
assays(x, withDimnames=TRUE, ...) &lt;- value
assay(x, i, withDimnames=TRUE, ...)
assay(x, i, withDimnames=TRUE, ...) &lt;- value
rowData(x, use.names=TRUE, ...)
rowData(x, ...) &lt;- value
colData(x, ...)
colData(x, ...) &lt;- value
#dim(x)
#dimnames(x)
#dimnames(x) &lt;- value

## Quick colData access

## S4 method for signature 'SummarizedExperiment'
x$name
## S4 replacement method for signature 'SummarizedExperiment'
x$name &lt;- value
## S4 method for signature 'SummarizedExperiment,ANY,missing'
x[[i, j, ...]]
## S4 replacement method for signature 'SummarizedExperiment,ANY,missing'
x[[i, j, ...]] &lt;- value

## Subsetting

## S4 method for signature 'SummarizedExperiment'
x[i, j, ..., drop=TRUE]
## S4 replacement method for signature 'SummarizedExperiment,ANY,ANY,SummarizedExperiment'
x[i, j] &lt;- value
## S4 method for signature 'SummarizedExperiment'
subset(x, subset, select, ...)

## Combining

## S4 method for signature 'SummarizedExperiment'
rbind(..., deparse.level=1)
## S4 method for signature 'SummarizedExperiment'
cbind(..., deparse.level=1)
## S4 method for signature 'SummarizedExperiment'
combineRows(x, ..., delayed=TRUE, fill=NA, use.names=TRUE)
## S4 method for signature 'SummarizedExperiment'
combineCols(x, ..., delayed=TRUE, fill=NA, use.names=TRUE)

## On-disk realization

## S4 method for signature 'SummarizedExperiment'
realize(x, BACKEND=getAutoRealizationBackend())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assays">assays</code></td>
<td>

<p>A <code>list</code> or <code>SimpleList</code> of matrix-like elements,
or a matrix-like object (e.g. an ordinary matrix, a data frame,
a <a href="../../S4Vectors/help/DataFrame.html">DataFrame</a> object from the <span class="pkg">S4Vectors</span> package,
a <a href="../../SparseArray/help/SparseMatrix.html">SparseMatrix</a> derivative from the <span class="pkg">SparseArray</span>
package, a <a href="../../Matrix/help/sparseMatrix.html">sparseMatrix</a> derivative from the <span class="pkg">Matrix</span>
package, a <a href="../../DelayedArray/help/DelayedMatrix.html">DelayedMatrix</a> object from the
<span class="pkg">DelayedArray</span> package, etc...).
All elements of the list must have the same dimensions, and dimension
names (if present) must be consistent across elements and with the row
names of <code>rowRanges</code> and <code>colData</code>.
</p>
</td></tr>
<tr><td><code id="rowData">rowData</code></td>
<td>

<p><code>NULL</code> (the default) or a <a href="../../S4Vectors/help/DataFrame.html">DataFrame</a> object
describing the rows. Row names, if present, become the row names of
the constructed SummarizedExperiment object. The number of rows of
the <a href="../../S4Vectors/help/DataFrame.html">DataFrame</a> must equal the number of rows of the
matrices in <code>assays</code>.
</p>
</td></tr>
<tr><td><code id="rowRanges">rowRanges</code></td>
<td>

<p><code>NULL</code> (the default), or a <a href="../../GenomicRanges/help/GRanges.html">GRanges</a> or
<a href="../../GenomicRanges/help/GRangesList.html">GRangesList</a> object describing the ranges of
interest.
If <code>NULL</code>, a SummarizedExperiment <em>instance</em> is returned.
Otherwise, a <a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> <em>instance</em> is returned.
Names on <code>rowRanges</code>, if present, become the row names of the
returned object. The length of the <a href="../../GenomicRanges/help/GRanges.html">GRanges</a> or
<a href="../../GenomicRanges/help/GRangesList.html">GRangesList</a> must equal the number of rows of
the matrices in <code>assays</code>.
</p>
</td></tr>
<tr><td><code id="colData">colData</code></td>
<td>

<p>An optional DataFrame describing the samples. Row names on
<code>colData</code>, if present, become the column names of the returned object.
</p>
</td></tr>
<tr><td><code id="metadata">metadata</code></td>
<td>

<p>An optional <code>list</code> of arbitrary content describing the overall
experiment.
</p>
</td></tr>
<tr><td><code id="checkDimnames">checkDimnames</code></td>
<td>

<p>By default the rownames and colnames of the supplied
assay(s) are checked for consistency with those of the
SummarizedExperiment or <a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> object
to construct. More precisely, the rownames and colnames of each assay
must be <code>NULL</code> or identical to those of the object.
Use <code>checkDimnames=FALSE</code> to skip this check.
</p>
</td></tr>
<tr><td><code id="x">x</code></td>
<td>

<p>A SummarizedExperiment object or derivative.
</p>
</td></tr>
<tr><td><code id="...">...</code></td>
<td>

<p>For <code>assay</code>, arguments in <code>...</code> are forwarded to <code>assays</code>.
</p>
<p>For <code>rbind</code>, <code>cbind</code>, <code>...</code> contains SummarizedExperiment
objects (or derivatives) to be combined.
</p>
<p>For other accessors, ignored.
</p>
</td></tr>
<tr><td><code id="value">value</code></td>
<td>

<p>An object of a class specified in the S4 method
signature or as outlined in &lsquo;Details&rsquo;.
</p>
</td></tr>
<tr><td><code id="i">i</code>, <code id="j">j</code></td>
<td>

<p>For <code>assay</code>, <code>assay&lt;-</code>, <code>i</code> is an integer or
numeric scalar; see &lsquo;Details&rsquo; for additional constraints.
</p>
<p>For <code>[,SummarizedExperiment</code>,
<code>[,SummarizedExperiment&lt;-</code>, <code>i</code>, <code>j</code> are subscripts
that can act to subset the rows and columns of <code>x</code>, that is the
<code>matrix</code> elements of <code>assays</code>.
</p>
<p>For <code>[[,SummarizedExperiment</code>,
<code>[[&lt;-,SummarizedExperiment</code>, <code>i</code> is a scalar index (e.g.,
<code>character(1)</code> or <code>integer(1)</code>) into a column of
<code>colData</code>.
</p>
</td></tr>
<tr><td><code id="name">name</code></td>
<td>

<p>A symbol representing the name of a column of <code>colData</code>.
</p>
</td></tr>
<tr><td><code id="withDimnames">withDimnames</code></td>
<td>

<p>A <code>logical(1)</code>, indicating whether the dimnames of the
SummarizedExperiment object (or derivative) should be applied (i.e.
copied) to the extracted assays. More precisely, setting
<code>withDimnames=FALSE</code> in the <em>getter</em> returns the
assays <em>as-is</em> whereas setting <code>withDimnames=TRUE</code>
return them with possibly modified dimnames.
</p>
<p>See &quot;Top-level dimnames vs assay-level dimnames&quot; section in
the SummarizedExperiment vignette for more information about
this and some examples.
</p>
<p>Setting <code>withDimnames=FALSE</code> in the <em>setter</em> (<code>assays&lt;-</code>)
is required when the dimnames on the supplied assays are not identical
to the dimnames on the SummarizedExperiment object; it does not
influence actual assignment of dimnames to assays (they're always
stored as-is).
</p>
<p>Note that </p>
<pre>  assays(x, withDimnames=FALSE) &lt;- assays(x, withDimnames=FALSE)</pre>
<p>is guaranteed to always work and be a no-op. This is not the case
if <code>withDimnames=TRUE</code> is used or if <code>withDimnames</code> is
not specified.
</p>
</td></tr>
<tr><td><code id="use.names">use.names</code></td>
<td>

<p>For <code>rowData</code>: Like <code><a href="../../S4Vectors/help/mcols.html">mcols</a>(x)</code>, by default
<code>rowData(x)</code> propagates the rownames of <code>x</code> to the returned
<a href="../../S4Vectors/help/DataFrame.html">DataFrame</a> object (note that for a SummarizedExperiment
object or derivative, the rownames are also the names i.e.
<code>rownames(x)</code> is always the same as <code>names(x)</code>).
Setting <code>use.names=FALSE</code> suppresses this propagation i.e. it
returns a <a href="../../S4Vectors/help/DataFrame.html">DataFrame</a> object with no rownames. Use this
when <code>rowData(x)</code> fails, which can happen when the rownames contain
NAs (because the rownames of a SummarizedExperiment object or derivative
can contain NAs, but the rownames of a <a href="../../S4Vectors/help/DataFrame.html">DataFrame</a> object
cannot).
</p>
<p>For <code>combineRows</code> and <code>combineCols</code>: See Combining section
below.
</p>
</td></tr>
<tr><td><code id="drop">drop</code></td>
<td>

<p>A <code>logical(1)</code>, ignored by these methods.
</p>
</td></tr>
<tr><td><code id="deparse.level">deparse.level</code></td>
<td>

<p>See <code>?base::<a href="../../base/help/cbind.html">cbind</a></code> for a description of this argument.
</p>
</td></tr>
<tr><td><code id="subset">subset</code></td>
<td>

<p>An expression which, when evaluated in the context of <code>rowData(x)</code>,
is a logical vector indicating elements or rows to keep: missing values
are taken as false.
</p>
</td></tr>
<tr><td><code id="select">select</code></td>
<td>

<p>An expression which, when evaluated in the context of <code>colData(x)</code>,
is a logical vector indicating elements or rows to keep: missing
values are taken as false.
</p>
</td></tr>
<tr><td><code id="delayed">delayed</code>, <code id="fill">fill</code></td>
<td>

<p>See <code>combineRows</code> and <code>combineCols</code> in Combining section
below.
</p>
</td></tr>
<tr><td><code id="BACKEND">BACKEND</code></td>
<td>

<p><code>NULL</code> (the default), or a single string specifying the
name of the backend. When the backend is set to <code>NULL</code>,
each element of <code>assays(x)</code> is realized in memory as an
ordinary array by just calling <code>as.array</code> on it.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SummarizedExperiment class is meant for numeric and other
data types derived from a sequencing experiment. The structure is
rectangular like a <code>matrix</code>, but with additional annotations on
the rows and columns, and with the possibility to manage several
assays simultaneously so long as they be of the same dimensions.
</p>
<p>The rows of a SummarizedExperiment object represent features
of interest. Information about these features is stored in a
DataFrame object, accessible using the function
<code>rowData</code>. The DataFrame must have as many rows
as there are rows in the SummarizedExperiment object, with each row
of the DataFrame providing information on the feature in the
corresponding row of the SummarizedExperiment object. Columns of the
DataFrame represent different attributes of the features
of interest, e.g., gene or transcript IDs, etc.
</p>
<p>Each column of a SummarizedExperiment object represents a sample.
Information about the samples are stored in a DataFrame,
accessible using the function <code>colData</code>, described below.
The DataFrame must have as many rows as there are
columns in the SummarizedExperiment object, with each row of the
DataFrame providing information on the sample in the
corresponding column of the SummarizedExperiment object.
Columns of the DataFrame represent different sample
attributes, e.g., tissue of origin, etc. Columns of the
DataFrame can themselves be annotated (via the
<code><a href="../../S4Vectors/help/mcols.html">mcols</a></code> function). Column names typically
provide a short identifier unique to each sample.
</p>
<p>A SummarizedExperiment object can also contain information about
the overall experiment, for instance the lab in which it was conducted,
the publications with which it is associated, etc. This information is
stored as a <code>list</code> object, accessible using the <code>metadata</code>
function. The form of the data associated with the experiment is left to
the discretion of the user.
</p>
<p>The SummarizedExperiment container is appropriate for matrix-like
data. The data are accessed using the <code>assays</code> function,
described below. This returns a SimpleList object. Each
element of the list must itself be a matrix (of any mode) and must
have dimensions that are the same as the dimensions of the
SummarizedExperiment in which they are stored. Row and column
names of each matrix must either be <code>NULL</code> or match those of the
SummarizedExperiment during construction. It is convenient for
the elements of SimpleList of assays to be named.
</p>


<h3>Accessors</h3>

<p>In the code snippets below, <code>x</code> is a SummarizedExperiment object
or derivative (e.g. a <a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> object).
</p>

<dl>
<dt><code>assays(x)</code>, <code>assays(x) &lt;- value</code>:</dt><dd><p>Get or set the
assays. <code>value</code> is a <code>list</code> or <code>SimpleList</code>, each
element of which is a matrix with the same dimensions as
<code>x</code>.</p>
</dd>
<dt><code>assay(x, i)</code>, <code>assay(x, i) &lt;- value</code>:</dt><dd><p>A convenient
alternative (to <code>assays(x)[[i]]</code>, <code>assays(x)[[i]] &lt;-
      value</code>) to get or set the <code>i</code>th (default first) assay
element. <code>value</code> must be a matrix of the same dimension as
<code>x</code>, and with dimension names <code>NULL</code> or consistent with
those of <code>x</code>.</p>
</dd>
<dt><code>assayNames(x)</code>, <code>assayNames(x) &lt;- value</code>:</dt><dd><p>Get or
set the names of <code>assay()</code> elements.</p>
</dd>
<dt><code>rowData(x, use.names=TRUE)</code>, <code>rowData(x) &lt;- value</code>:</dt><dd>
<p>Get or set the row data. <code>value</code> is a DataFrame object.</p>
</dd>
<dt><code>colData(x)</code>, <code>colData(x) &lt;- value</code>:</dt><dd><p>Get or set the
column data. <code>value</code> is a DataFrame object. Row
names of <code>value</code> must be <code>NULL</code> or consistent with the
existing column names of <code>x</code>.</p>
</dd>
<dt><code>metadata(x)</code>, <code>metadata(x) &lt;- value</code>:</dt><dd><p>Get or set
the experiment data. <code>value</code> is a <code>list</code> with arbitrary
content.</p>
</dd>
<dt><code>dim(x)</code>:</dt><dd><p>Get the dimensions (features of interest x samples)
of the SummarizedExperiment object.</p>
</dd>
<dt><code>dimnames(x)</code>, <code>dimnames(x) &lt;- value</code>:</dt><dd><p>Get or set
the dimension names. <code>value</code> is usually a list of length 2,
containing elements that are either <code>NULL</code> or vectors of
appropriate length for the corresponding dimension. <code>value</code>
can be <code>NULL</code>, which removes dimension names. This method
implies that <code>rownames</code>, <code>rownames&lt;-</code>, <code>colnames</code>,
and <code>colnames&lt;-</code> are all available.</p>
</dd>
</dl>



<h3>Subsetting</h3>

<p>In the code snippets below, <code>x</code> is a SummarizedExperiment object
or derivative (e.g. a <a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> object).
</p>

<dl>
<dt><code>x[i,j]</code>, <code>x[i,j] &lt;- value</code>:</dt><dd><p>Create or replace a
subset of <code>x</code>. <code>i</code>, <code>j</code> can be <code>numeric</code>,
<code>logical</code>, <code>character</code>, or <code>missing</code>. <code>value</code>
must be a SummarizedExperiment object with dimensions,
dimension names, and assay elements consistent with the subset
<code>x[i,j]</code> being replaced.</p>
</dd>
<dt><code>subset(x, subset, select)</code>:</dt><dd><p>Create a subset of <code>x</code>
using an expression <code>subset</code> referring to columns of
<code>rowData(x)</code> and / or <code>select</code> referring to column names
of <code>colData(x)</code>.</p>
</dd>
</dl>

<p>Additional subsetting accessors provide convenient access to
<code>colData</code> columns
</p>

<dl>
<dt><code>x$name</code>, <code>x$name &lt;- value</code></dt><dd><p>Access or replace
column <code>name</code> in <code>x</code>.</p>
</dd>
<dt><code>x[[i, ...]]</code>, <code>x[[i, ...]] &lt;- value</code></dt><dd><p>Access or
replace column <code>i</code> in <code>x</code>.</p>
</dd>
</dl>



<h3>Combining</h3>

<p>In the code snippets below, <code>x</code>, <code>y</code> and <code>...</code> are
SummarizedExperiment objects (or derivatives) to be combined.
</p>

<dl>
<dt><code>rbind(...)</code>:</dt><dd>
<p><code>rbind</code> combines objects with the same samples
but different features of interest (rows in <code>assays</code>).
The colnames in <code>rowData(SummarizedExperiment)</code> must match or
an error is thrown.
Duplicate columns of <code>colData(SummarizedExperiment)</code> must
contain the same data.
</p>
<p>Data in <code>assays</code> are combined by name matching; if all assay
names are <code>NULL</code> matching is by position. A mixture of names
and <code>NULL</code> throws an error.
</p>
<p><code>metadata</code> from all objects are combined into a <code>list</code>
with no name checking.
</p>
</dd>
<dt><code>cbind(...)</code>:</dt><dd>
<p><code>cbind</code> combines objects with the same features of interest
but different samples (columns in <code>assays</code>).
The colnames in <code>colData(SummarizedExperiment)</code> must match or
an error is thrown.
Duplicate columns of <code>rowData(SummarizedExperiment)</code> must
contain the same data.
</p>
<p>Data in <code>assays</code> are combined by name matching; if all assay
names are <code>NULL</code> matching is by position. A mixture of names
and <code>NULL</code> throws an error.
</p>
<p><code>metadata</code> from all objects are combined into a <code>list</code>
with no name checking.
</p>
</dd>
<dt><code>combineRows(x, ..., use.names=TRUE, delayed=TRUE, fill=NA)</code>:</dt><dd>
<p><code>combineRows</code> acts like more flexible <code>rbind</code>, returning a
SummarizedExperiment with features equal to the concatenation of features
across all input objects. Unlike <code>rbind</code>, it permits differences in
the number and identity of the columns, differences in the available
<code><a href="../../SummarizedExperiment/help/rowData.html">rowData</a></code> fields, and even differences in the available
<code><a href="../../SummarizedExperiment/help/assays.html">assays</a></code> among the objects being combined.
</p>
<p>If <code>use.names=TRUE</code>, each input object must have non-<code>NULL</code>,
non-duplicated column names. These names do not have to be the same, or
even shared, across the input objects. The column names of the returned
<code>SummarizedExperiment</code> will be a union of the column names across
all input objects. If a column is not present in an input, the
corresponding assay and <code>colData</code> entries will be filled with
<code>fill</code> and <code>NA</code>s, respectively, in the combined
SummarizedExperiment.
</p>
<p>If <code>use.names=FALSE</code>, all objects must have the same number of
columns. The column names of the returned object is set to
<code>colnames(x)</code>. Any differences in the column names between input
objects are ignored.
</p>
<p>Data in <code>assays</code> are combined by matching the names of the assays.
If one input object does not contain a named assay present in other input
objects, the corresponding assay entries in the returned object will be
set to <code>fill</code>. If all assay names are <code>NULL</code>, matching is done
by position. A mixture of named and unnamed assays will throw an error.
</p>
<p>If <code>delayed=TRUE</code>, assay matrices are wrapped in
<code>DelayedArray</code>s to avoid any extra memory allocation during
the matrix <code>rbind</code>ing. Otherwise, the matrices are combined as-is;
note that this may still return <code>DelayedMatrix</code>s if the inputs were
also <code>DelayedMatrix</code> objects.
</p>
<p>If any input is a <code>RangedSummarizedExperiment</code>, the returned object
will also be a <code>RangedSummarizedExperiment</code>. The <code>rowRanges</code> of
the returned object is set to the concatenation of the <code>rowRanges</code>
of all inputs. If any input is a <code>SummarizedExperiment</code>, the
returned <code>rowRanges</code> is converted into a <code>GRangesList</code> and the
entries corresponding to the rows of the <code>SummarizedExperiment</code> are
set to zero-length <code>GRanges</code>. If all inputs are
<code>SummarizedExperiment</code> objects, a <code>SummarizedExperiment</code> is
also returned.
</p>
<p><code>rowData</code> are combined using <code><a href="../../SummarizedExperiment/help/combineRows.html">combineRows</a></code> for
<code>DataFrame</code> objects. It is not necessary for all input objects to
have the same fields in their <code>rowData</code>; missing fields are filled
with <code>NA</code>s for the corresponding rows in the returned object.
</p>
<p><code>metadata</code> from all objects are combined into a <code>list</code>
with no name checking.
</p>
</dd>
<dt><code>combineCols(x, ..., use.names=TRUE, delayed=TRUE, fill=NA)</code>:</dt><dd>
<p><code>combineCols</code> acts like more flexible <code>cbind</code>, returning a
SummarizedExperiment with columns equal to the concatenation of columns
across all input objects. Unlike <code>cbind</code>, it permits differences in
the number and identity of the rows, differences in the available
<code><a href="../../SummarizedExperiment/help/colData.html">colData</a></code> fields, and even differences in the available
<code><a href="../../SummarizedExperiment/help/assays.html">assays</a></code> among the objects being combined.
</p>
<p>If <code>use.names=TRUE</code>, each input object must have non-<code>NULL</code>,
non-duplicated row names. These names do not have to be the same, or
even shared, across the input objects. The row names of the returned
<code>SummarizedExperiment</code> will be a union of the row names across
all input objects. If a row is not present in an input, the
corresponding assay and <code>rowData</code> entries will be filled with
<code>fill</code> and <code>NA</code>s, respectively, in the combined
SummarizedExperiment.
</p>
<p>If <code>use.names=FALSE</code>, all objects must have the same number of rows.
The row names of the returned object is set to <code>rownames(x)</code>. Any
differences in the row names between input objects are ignored.
</p>
<p>Data in <code>assays</code> are combined by matching the names of the assays.
If one input object does not contain a named assay present in other input
objects, the corresponding assay entries in the returned object will be
set to <code>fill</code>. If all assay names are <code>NULL</code>, matching is
done by position. A mixture of named and unnamed assays will throw an
error.
</p>
<p>If <code>delayed=TRUE</code>, assay matrices are wrapped in
<code>DelayedArray</code>s to avoid any extra memory allocation during
the matrix <code>rbind</code>ing. Otherwise, the matrices are combined as-is;
note that this may still return <code>DelayedMatrix</code>s if the inputs were
also <code>DelayedMatrix</code> objects.
</p>
<p>If any input is a <code>RangedSummarizedExperiment</code>, the returned object
will also be a <code>RangedSummarizedExperiment</code>. The <code>rowRanges</code> of
the returned object is set to a merge of the <code>rowRanges</code> of all
inputs, where the coordinates for each row are taken from the input
object that contains that row. Any conflicting ranges for shared rows
will raise a warning and all <code>rowRanges</code> information from the
offending <code>RangedSummarizedExperiment</code> will be ignored. If any
input is a <code>SummarizedExperiment</code>, the returned <code>rowRanges</code> is
converted into a <code>GRangesList</code> and the entries corresponding to the
unique rows of the <code>SummarizedExperiment</code> are set to zero-length
<code>GRanges</code>. If all inputs are <code>SummarizedExperiment</code> objects, a
<code>SummarizedExperiment</code> is also returned.
</p>
<p><code>colData</code> are combined using <code><a href="../../SummarizedExperiment/help/combineRows.html">combineRows</a></code> for
<code>DataFrame</code> objects. It is not necessary for all input objects to
have the same fields in their <code>colData</code>; missing fields are filled
with <code>NA</code>s for the corresponding columns in the returned object.
</p>
<p><code>metadata</code> from all objects are combined into a <code>list</code>
with no name checking.
</p>
</dd>
</dl>



<h3>Implementation and Extension</h3>

<p>This section contains advanced material meant for package developers.
</p>
<p>SummarizedExperiment is implemented as an S4 class, and can be extended in
the usual way, using <code>contains="SummarizedExperiment"</code> in the new
class definition.
</p>
<p>In addition, the representation of the <code>assays</code> slot of
SummarizedExperiment is as a virtual class Assays. This
allows derived classes (<code>contains="Assays"</code>) to implement
alternative requirements for the assays, e.g., backed by file-based
storage like NetCDF or the <code>ff</code> package, while re-using the existing
SummarizedExperiment class without modification.
See <a href="../../SummarizedExperiment/help/Assays.html">Assays</a> for more information.
</p>


<h3>Author(s)</h3>

<p>Martin Morgan; <code>combineRows</code> and <code>combineCols</code> by Aaron Lun</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="../../SummarizedExperiment/help/RangedSummarizedExperiment.html">RangedSummarizedExperiment</a> objects.
</p>
</li>
<li> <p><a href="../../S4Vectors/help/DataFrame.html">DataFrame</a>, <a href="../../S4Vectors/help/SimpleList.html">SimpleList</a>, and
<a href="../../S4Vectors/help/Annotated.html">Annotated</a> objects in the <span class="pkg">S4Vectors</span> package.
</p>
</li>
<li><p> The <code><a href="../../S4Vectors/help/metadata.html">metadata</a></code> and
<code><a href="../../S4Vectors/help/mcols.html">mcols</a></code> accessors in the <span class="pkg">S4Vectors</span>
package.
</p>
</li>
<li> <p><code><a href="../../HDF5Array/help/saveHDF5SummarizedExperiment.html">saveHDF5SummarizedExperiment</a></code> and
<code><a href="../../HDF5Array/help/loadHDF5SummarizedExperiment.html">loadHDF5SummarizedExperiment</a></code> in the
<span class="pkg">HDF5Array</span> package for saving/loading an HDF5-based
SummarizedExperiment object to/from disk.
</p>
</li>
<li><p> The <code><a href="../../DelayedArray/help/realize.html">realize</a></code> generic function in the
<span class="pkg">DelayedArray</span> package for more information about on-disk
realization of objects carrying delayed operations.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>nrows &lt;- 200; ncols &lt;- 6
counts &lt;- matrix(runif(nrows * ncols, 1, 1e4), nrows)
colData &lt;- DataFrame(Treatment=rep(c("ChIP", "Input"), 3),
                     row.names=LETTERS[1:6])
se0 &lt;- SummarizedExperiment(assays=SimpleList(counts=counts),
                            colData=colData)
se0
dim(se0)
dimnames(se0)
assayNames(se0)
head(assay(se0))
assays(se0) &lt;- endoapply(assays(se0), asinh)
head(assay(se0))

rowData(se0)
colData(se0)

se0[, se0$Treatment == "ChIP"]
subset(se0, select = Treatment == "ChIP")

## rbind() combines objects with the same samples but different
## features of interest:
se1 &lt;- se0
se2 &lt;- se1[1:50,]
rownames(se2) &lt;- letters[seq_len(nrow(se2))]
cmb2 &lt;- rbind(se1, se2)
dim(cmb2)
dimnames(cmb2)

## cbind() combines objects with the same features of interest
## but different samples:
se1 &lt;- se0
se2 &lt;- se1[,1:3]
colnames(se2) &lt;- letters[seq_len(ncol(se2))]
cmb1 &lt;- cbind(se1, se2)
dim(cmb1)
dimnames(cmb1)

## ---------------------------------------------------------------------
## ON-DISK REALIZATION
## ---------------------------------------------------------------------
library(DelayedArray)
setAutoRealizationBackend("HDF5Array")
cmb3 &lt;- realize(cmb2)
assay(cmb3, withDimnames=FALSE)  # an HDF5Matrix object
</code></pre>

<hr><div style="text-align: center;">[Package <em>SummarizedExperiment</em> version 1.40.0 <a href="00Index.html">Index</a>]</div></main>

</div>
</body></html>
