<!DOCTYPE html><html><head><title>R: Plot exons of differentially expressed gene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script>
const macros = { "\\R": "\\textsf{R}", "\\mbox": "\\text", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css">
</head><body><div class="container"><main>

<table style="width: 100%;"><tr><td>plotExons {limma}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Plot exons of differentially expressed gene</h2>

<h3>Description</h3>

<p>Plot exons of differentially expressed gene and mark the differentially expressed exons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotExons(fit, coef = ncol(fit), geneid = NULL, genecolname = "GeneID",
          exoncolname = NULL, rank = 1L, FDR = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit">fit</code></td>
<td>
<p><code>MArrayLM</code> fit object produced by <code>eBayes</code>.</p>
</td></tr>
<tr><td><code id="coef">coef</code></td>
<td>
<p>the coefficient (column) of fit for which differential expression is assessed.</p>
</td></tr>
<tr><td><code id="geneid">geneid</code></td>
<td>
<p>character string, ID of the gene to plot.</p>
</td></tr>
<tr><td><code id="genecolname">genecolname</code></td>
<td>
<p>character string for the column name of <code>fit$genes</code> containing gene IDs. Defaults to &quot;GeneID&quot; for Entrez Gene ID.</p>
</td></tr>
<tr><td><code id="exoncolname">exoncolname</code></td>
<td>
<p>character string for the column name of <code>fit$genes</code> containing exon IDs.</p>
</td></tr>
<tr><td><code id="rank">rank</code></td>
<td>
<p>integer, if <code>geneid=NULL</code> then this ranked gene will be plotted.</p>
</td></tr>
<tr><td><code id="FDR">FDR</code></td>
<td>
<p>numeric, mark differentially expressed exons with false discovery rate less than this cutoff.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots log2-fold-change by exon for the specified gene and highlight the differentially expressed exons.
Show annotations such as GeneID, Symbol and Strand if available as title for the gene to plot.
The significantly differentially expressed individual exons are highlighted as red dots for up-regulation and as blue dots for down-regulation.
The size of the dots are weighted by its significance.
</p>


<h3>Value</h3>

<p>A plot is created on the current graphics device.</p>


<h3>Author(s)</h3>

<p>Yifang Hu and Gordon Smyth</p>


<h3>See Also</h3>

<p><code><a href="../../limma/help/lmFit.html">lmFit</a></code>, <code><a href="../../limma/help/eBayes.html">eBayes</a></code>, <code><a href="../../limma/help/plotSplice.html">plotSplice</a></code>
</p>
<p>A summary of functions available in LIMMA for RNA-seq analysis is given in <a href="../../limma/help/11.RNAseq.html">11.RNAseq</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
fit &lt;- lmFit(y,design)
fit &lt;- eBayes(fit)
plotExons(fit)
plotExons(fit, exoncolname = "Start", rank = 1)
plotExons(fit, geneid = "ps", genecolname = "Symbol", exoncolname = "Start")

## End(Not run)

</code></pre>

<hr><div style="text-align: center;">[Package <em>limma</em> version 3.66.0 <a href="00Index.html">Index</a>]</div></main>

</div>
</body></html>
